lebedigital/openbis/interbis.py

```diff
@@ -0,0 +1,495 @@
+                df = df.join(prop_df)
+
+        df = df.transpose().reset_index(drop=True)
+        df = df.fillna(value="")
+        return df
+
+    def get_sample_dict(self, identifier: str) -> dict:
+        """Fetches a dictionary filled with information about the sample
+
+        Very useful method actually if you already have the identifier. You can get all information about
+        the sample from it like all metadata, collection, type, etc.
+
+        Args:
+            identifier (str): Identifier of the sample (identifier or permID)
+
+        Returns:
+            dict: dict containing all information about the sample
+        """
+        sample = self.get_sample(identifier)
+        sample_info_dict = sample.attrs.all()
+        sample_prop_dict = {key.upper(): val for key, val in sample.p.all().items()}
+        full_dict = (sample_info_dict | sample_prop_dict)
+
+        # copying the information from key experiment to collection for ease of use
+        full_dict['collection'] = full_dict.get('experiment')
+        return full_dict
+
+    def get_sample_identifier(self, name: str) -> str:
+        """
+
+        Returns the full identifier of the sample
+
+        Args:
+            name (str): '$name' attribute of the sample
+
+        Returns:
+            str: Identifier of sample
+        """
+        sample_df = self.get_samples(where={'$name': name}).df
+        if len(sample_df.index) > 1 or len(sample_df.index) == 0:
+            raise ValueError(
+                f'Could not find unique sample, the amount of samples with that name is {len(sample_df.index)}')
+
+        return sample_df['identifier'].values[0]
+
+    def get_dataset_identifier(self, name: str) -> list[str]:
+        """
+
+        Returns the permId of the dataset
+
+        Args:
+            name (str): '$name' attribute of the dataset
+
+        Returns:
+            list[str]: List of the dataset identifiers
+        """
+        df = self.get_datasets(**{"$NAME": name}).df
+        if len(df) == 0:
+            logging.warning("No dataset found")
+            return []
+        elif len(df) > 1:
+            logging.warning(f"More than one dataset with the name {name} found")
+
+        return df['permId'].to_list()
+
+    def get_collection_identifier(self, collection_code: str) -> str:
+        """Returns the full identifier of the collection
+
+        Args:
+            collection_code (str): Name of the collection
+
+        Returns:
+            str: Identifier of the collection
+        """
+        collections_df = self.get_collections(props='$name').df
+        collection_code = collection_code.upper()
+
+        if len(collections_df.index):
+            return collections_df[collections_df['identifier'].str.contains(
+                collection_code)].iloc[0]['identifier']
+        else:
+            raise ValueError(f'No collection with name {collection_code} found')
+
+    def exists_in_datastore(self, name: str) -> bool:
```

Comment:
should the name reflect the checking for samples (instead of e.g. collections?), in_datastore is may not required (if you would like to shorten the name), since this class is only interfacing the openBIS instance and IMO there is no other option

Reply:
Yes that is true but while naming it I found the name "exists" a bit ambiguous. I have left it as is for now.